src/tools/ScenarioMeasurement/Startup/Startup.cs

```diff
@@ -314,5 +327,19 @@ private static void WriteResultTable(IEnumerable<Counter> counters, Logger logge
                 logger.Log($"{counter.Name,-15}|{average,-15}|{min,-15}|{max,-15}");
             }
         }
+
+        private static Dictionary<string, string> ParseStringToDictionary(string s)
+        {
+            Dictionary<string, string> dict = new Dictionary<string, string>();
+            foreach (string substring in s.Split(';', StringSplitOptions.RemoveEmptyEntries))
+            {
+                string key = substring.Split('=').First();
+                string value = substring.Split('=').Last();
```

Comment:
Instead of calling split twice, store the array and index into it.
```
var values = substring.Split('=');
var key = values[0];
var value = values[1];
```
`Split` will also naturally give you the empty string, so nothing has to be done to handle the case I called out below:
```
> "foo=".Split('=')
string[2] { "foo", "" }
```

Reply:
nice! Didn't know it could gently handle index out of bounds.